### ISSUE TITLE
Marx Generator will not form

### ISSUE THREAD

**Issue by generilisk** [malte0811/IndustrialWires#15]
- What is the bug? Marx Generator will not form
- What did you do to make it happen? Built as following illustration in IE guide, no result upon using hammer to form multiblock.
- Does it happen every time you do this? Yes.
- Did this happen on a dedicated server (multiplayer servers), in LAN multiplayer or in singleplayer? Singleplayer world in creative and survival
- What other mods were installed when the bug happened? This is in the pack "All the Mods 3" version 3.3

**Comment by malte0811** [comment]
I'm currently downloading ATM.
This may or may not be related: What happened to the Marx generator in the back of the second picture? This shouldn't happen in the released version.
For reference: [Screenshots from Curse](https://imgur.com/a/rAVBD).

**Comment by malte0811** [comment]
I just tested in ATM and it worked just fine for me. BTW, you do need 8 fences to build a Marx Generator. The 7 in the manual is a typo.

**Comment by generilisk** [comment]
The background one was placed with the auto-place feature available in creative. It spawned like that.

**Comment by generilisk** [comment]
Messing around in the test world, I just figured this out. The illustration in the engineer's manual is mirrored from what it should be. The redstone wire connector needs to be on the right, and the HV connector needs to be on the left. The lower fences need to be on the right, and the higher fences need to be on the left.
In addition, the Engineer's Manual says 7 Steel Fence are required, but the structure actually requires 8.

**Comment by Zygus42** [comment]
forge-1.12.2-14.23.0.2491-installer-win
IndustrialWires-1.6-20
ImmersiveEngineering-0.12-72
industrialcraft-2-2.8.16-ex112
![2017-10-08_01 12 57](https://user-images.githubusercontent.com/20757445/31310182-f05047e0-abc5-11e7-87f8-499e6779ccd7.png)

I also cannot get this to form, and I tried reversing the sides as above comment, as well as then reversing the way the steel walllmounts were facing, but again nothing forms. Clicking on any of the capacitors with the hammer just changes their input/output state for that side.  Clicking any other of the blocks from any of the sides also does nothing, except for the redstone wire connector which changes state.

Tried in multiplayer survival server as well as singleplayer creative.

**Comment by malte0811** [comment]
You need to connect the HV relays in vertical columns, each one to the one directly above it. The rendering in the manual for that is kind of broken, no idea why.. I formed one just a few hours ago on the BTMMoon server, so it works in principle. Wires connected to the Marx generator may not render on dedicated servers (but will still work), fixed for the next release. And if you get it working: Either wear earmuffs/ear defenders, have a bucket of milk ready or don't stand too close to the MG.

**Comment by Zygus42** [comment]
yep - just got back to my real game and realised i had 8 HV wires left over - connected them to the reversed machine and it worked !

**Comment by Zygus42** [comment]
As it was now backwards from what i wanted in terms of input, i took it back apart, and it lost the 8 HV wires, which was a bit of an expensive pain.

**Comment by Zygus42** [comment]
So, I still can't get it to go.  Hooked it up to a power source and added a redstone cable to the port. See picture:
![2017-10-08_02 43 49](https://user-images.githubusercontent.com/20757445/31310881-cd1ee512-abd2-11e7-9efb-106b93de3b94.png)

I tested the power with a forestry machine and the redstone cabling with a redstone lamp, and they both work, and I'm sure that the capacitor input/output are correct.  Is there anything else that needs to be done ?

**Comment by malte0811** [comment]
Running the MG is more complicated than just giving it an RS signal on the white channel. White and yellow set the charging voltage, your charging voltage is a bit (~5kV) below the voltage needed to make it trigger by itself. To get it to discharge you can either connect a high RS signal to the yellow channel (increasing the charging voltage) or give it a pulse on the light blue channel (the firing control). If you want to use it for ore processing I'd suggest using control panel(s) to run and monitor the MG.
In general the MG is not something you can use properly without reading the whole manual entry (except possibly Appendix A). To use it to process ores you need to calculate the energy stored in it and, if you want the absolute maximum output, do some testing yourself.

**Comment by theArhar** [comment]
I tried everything and cant understand how exactly do I control this thing
I build control panel this type:
![2018-01-27_17 03 38](https://user-images.githubusercontent.com/8017191/35472211-058735ca-037c-11e8-947c-f03244faddf5.png)
And tryed like every position, waiting to fully charge, not waiting
And closest to working I can get is 8 white 0 yellow fully charge from 1 Iron ore 2 Iron nuggets
Please do an example or tutorial!

**Comment by theArhar** [comment]
Also, it forms only mirror-style from book

**Comment by malte0811** [comment]
You can't process a single piece of iron ore efficiently, the ideal voltage for that is too low. For two blocks of iron ore 10 white, 1 yellow is ideal. You may have to vary the value a little (up to 7 yellow units in either direction, but probably a lot less than that). You can calculate these values with the formulas in Appendix A. I will be adding a block that allows checking the amount of energy in the next release for people who don't want to do the maths.
EDIT: forming works perfectly fine for me, both mirrored and non-mirrored. Did you swap the HV and redstone connectors in the back as well (you should swap them)?

**Comment by theArhar** [comment]
Yes, NOW I understand the math
1. Take manual
2. Read Ideal Energy number (for example uranium 125)
3. Multiply it by 250 (31250)
4. Sqrt it (176,7)
5. Divide it by 250 and multiply 255 (180)
6. Divide by 16 (11) ==WHITE
7. Get reminder (4) ==Yellow
8. Place 1 uranium ore
9. Profit

And yes, it forms both ways - I just clicked on wrong side while trying to form it, thats why it formed only one way for me
